node-graph/gcore/src/vector/vector_attributes.rs

```diff
@@ -678,6 +679,44 @@ impl FoundSubpath {
 	pub fn contains(&self, segment_id: SegmentId) -> bool {
 		self.edges.iter().any(|s| s.id == segment_id)
 	}
+
+	pub fn to_bezpath(&self, vector: &Vector) -> kurbo::BezPath {
+		let mut bezpath = kurbo::BezPath::new();
+
+		if let Some(first_edge) = self.edges.first() {
+			let start_pos = vector.point_domain.positions()[first_edge.start];
+			bezpath.move_to(dvec2_to_point(start_pos));
+		}
+
+		for edge in &self.edges {
+			let segment_index = vector.segment_domain.ids().iter().position(|&id| id == edge.id).expect("Segment ID must exist");
```

Comment:
Please try to avoid O(n^2) algorithms when you could just store the segment index in the `HalfEdge` struct.